test/e2e/run.sh

```diff
@@ -12,8 +12,8 @@ set -euo pipefail
 for PKG in $(go list github.com/elastic/cloud-on-k8s/test/e2e/...); do
     if [ "$E2E_JSON" == "true" ]
     then
-        go test -v -failfast -timeout=2h -tags=e2e -p=1 --json $PKG $@
+        go test -v -failfast -timeout=4h -tags=e2e -p=1 --json $PKG $@
```

Comment:
Should we reach go test timeout before Jenkins timeout to get better test output?

Reply:
:+1: I did that, but it's still not guaranteed as `go test` timeout is per package and Jenkins one is overall.